[expo-notifications][bare] Android push handling crashes due to SoundResolver

## 🐛 Bug Report

@sjchmiela I'm seeing this crash in my bare app when upgrading expo-notifications from 1.2 -> 1.4. I think the issue was introduced when you added support for custom sound on android. Now, I can't get the push to work if I dont have a custom sound configured (i.e. I just want the default sound)

### Environment

```

  Expo CLI 3.20.1 environment info:
    System:
      OS: macOS 10.15.3
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.14.0 - ~/.nodenv/versions/12.14.0/bin/node
      Yarn: 1.12.1 - ~/.yarn/bin/yarn
      npm: 6.13.4 - ~/.nodenv/versions/12.14.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      @expo/webpack-config: 0.11.17 => 0.11.17 
      expo: ^37.0.0 => 37.0.3 
      react: ~16.9.0 => 16.9.0 
      react-dom: 16.9.0 => 16.9.0 
      react-native: ~0.61.5 => 0.61.5 
      react-native-web: ^0.11.7 => 0.11.7 
```

Bare expo

### Steps to Reproduce

Send a push notification to android via expo push tool with expo-notifications@1.4.0.


```js
fetch("https://exp.host/--/api/v2/push/send", {
  "headers": {
    "accept": "application/json",
    "accept-language": "en-US,en;q=0.9,es-US;q=0.8,es;q=0.7",
    "content-type": "application/json",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "cross-site"
  },
  "referrer": "https://expo.io/",
  "referrerPolicy": "origin",
  "body": "[{\"to\":\"ExponentPushToken[...]\",\"title\":\"test\",\"body\":\"this\",\"sound\":\"default\",\"data\":{},\"channelId\":\"bedtime-checklist\"}]",
  "method": "POST",
  "mode": "cors",
  "credentials": "omit"
});
```

### Expected Behavior

Push is displayed in notification tray

### Actual Behavior

```
05-04 14:11:55.223 29457 29526 E AndroidRuntime: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at java.lang.String.substring(String.java:2064)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at expo.modules.notifications.notifications.SoundResolver.resolve(SoundResolver.java:35)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at expo.modules.notifications.notifications.JSONNotificationContentBuilder.getSound(JSONNotificationContentBuilder.java:109)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at expo.modules.notifications.notifications.JSONNotificationContentBuilder.shouldPlayDefaultSound(JSONNotificationContentBuilder.java:96)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at expo.modules.notifications.notifications.JSONNotificationContentBuilder.setPayload(JSONNotificationContentBuilder.java:41)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at expo.modules.notifications.FirebaseListenerService.createNotification(FirebaseListenerService.java:94)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at expo.modules.notifications.FirebaseListenerService.onMessageReceived(FirebaseListenerService.java:85)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at com.google.firebase.messaging.FirebaseMessagingService.zzc(com.google.firebase:firebase-messaging@@20.1.0:78)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at com.google.firebase.messaging.zze.run(com.google.firebase:firebase-messaging@@20.1.0:2)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:6)
05-04 14:11:55.223 29457 29526 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)
```

### Reproducible Demo

Can work on a minimal repro if the issue isn't apparent.

<!--
  Please share a project that reproduces the issue.
  There are two ways to do it:

    * Create a new app using https://snack.expo.io/ and try to reproduce the issue in it.
      This is useful if you roughly know where the problem is, or can’t share the real code.

    * Or, copy your app and remove things until you’re left with the minimal reproducible demo.
      This is useful for finding the root cause. You may then optionally create a Snack.

  This is a good guide to creating bug demos: https://stackoverflow.com/help/mcve
  Once you’re done, copy and paste the link to the Snack or a public GitHub repository below.

  (Even if your issue can only be reproduced in a standalone app, please include a Snack
  or GitHub repo we can use to build a standalone app!)
-->

<!--
  What happens if you skip this step?

  Someone will read your bug report, and maybe will be able to help you,
  but it’s unlikely that it will get much attention from the team. Eventually,
  the issue will likely get closed in favor of issues that have reproducible demos.

  Please remember that:

    * Issues without reproducible demos have a very low priority.
    * The person fixing the bug would have to do that anyway. Please be respectful of their time.
    * You might figure out the issues yourself as you work on extracting it.

  Thanks for helping us help you!
-->

If I modify the native code to always useDefaultSound() then I get the following stacktrace when receiving the notification while the app is running:

```
05-04 16:30:56.215 11331 11360 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()                                                     
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at android.os.AsyncTask$4.done(AsyncTask.java:399)                                                                          
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)                                                    
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)                                                        
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:271)                                                                 
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)                                          
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)                                          
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)                                                                                    
05-04 16:30:56.215 11331 11360 E AndroidRuntime: Caused by: java.lang.UnsupportedOperationException: Could not put a value of class org.json.JSONObject$1 to bundle.                
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at org.unimodules.core.arguments.MapArguments.toBundle(MapArguments.java:160)                                               
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.NotificationSerializer.toBundle(NotificationSerializer.java:107)                
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.NotificationSerializer.toBundle(NotificationSerializer.java:60)                 
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.NotificationSerializer.toBundle(NotificationSerializer.java:47)                 
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.NotificationSerializer.toBundle(NotificationSerializer.java:39)                 
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.handling.SingleNotificationHandlerTask.start(SingleNotificationHandlerTask.java$
76)                                                                                                                                                                                 
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.handling.NotificationsHandler.onNotificationReceived(NotificationsHandler.java:$
16)                                                                                                                                                                                 
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.NotificationManager.onNotificationReceived(NotificationManager.java:83)         
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.service.ExpoNotificationsService.onNotificationReceived(ExpoNotificationsServic$
.java:106)                                                                                                                                                                          
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at expo.modules.notifications.notifications.service.BaseNotificationsService.onHandleWork(BaseNotificationsService.java:193$
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)                            
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)                            
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at android.os.AsyncTask$3.call(AsyncTask.java:378)                                                                          
05-04 16:30:56.215 11331 11360 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
```

--- a/apps/test-suite/tests/NewNotifications.js
+++ b/apps/test-suite/tests/NewNotifications.js
@@ -764,7 +764,7 @@ export async function test(t) {
         10000
       );
 
-      t.fit(
+      t.it(
         'triggers a notification which contains the custom color',
         async () => {
           const notificationReceivedSpy = t.jasmine.createSpy('notificationReceived');

--- a/packages/expo-notifications/CHANGELOG.md
+++ b/packages/expo-notifications/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fixed `SoundResolver` causing crash if the `sound` property is not defined or doesn't contain a `.` ([#8150](https://github.com/expo/expo/pull/8150) by [@sjchmiela](https://github.com/sjchmiela))
+
 ## [0.1.4] - 2020-05-04
 
 ### 🛠 Breaking changes

--- a/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/JSONNotificationContentBuilder.java
+++ b/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/JSONNotificationContentBuilder.java
@@ -105,8 +105,13 @@ public class JSONNotificationContentBuilder extends NotificationContent.Builder 
       // it's not a boolean, let's handle it as a string
     }
 
-    String soundValue = payload.optString(SOUND_KEY);
-    return mSoundResolver.resolve(soundValue);
+    try {
+      String soundValue = payload.getString(SOUND_KEY);
+      return mSoundResolver.resolve(soundValue);
+    } catch (JSONException e) {
+      // if it's neither a boolean nor a string we can't handle it
+    }
+    return null;
   }
 
   @Nullable

--- a/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/SoundResolver.java
+++ b/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/SoundResolver.java
@@ -32,7 +32,7 @@ public class SoundResolver {
     }
 
     String packageName = mContext.getPackageName();
-    String resourceName = filename.substring(0, filename.lastIndexOf('.'));
+    String resourceName = filenameToBasename(filename);
     int resourceId = mContext.getResources().getIdentifier(resourceName, "raw", packageName);
     // If resourceId is 0, then the resource does not exist.
     // Returning null falls back to using a default sound.
@@ -46,5 +46,13 @@ public class SoundResolver {
     }
 
     return Settings.System.DEFAULT_NOTIFICATION_URI;
+  }
+
+  private String filenameToBasename(String filename) {
+    if (!filename.contains(".")) {
+      return filename;
+    }
+
+    return filename.substring(0, filename.lastIndexOf('.'));
   }
 }

--- a/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/SoundResolver.java
+++ b/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/SoundResolver.java
@@ -27,7 +27,7 @@ public class SoundResolver {
    */
   @Nullable
   public Uri resolve(@Nullable String filename) {
-    if (filename == null) {
+    if (filename == null || filename.length() == 0) {
       return null;
     }
 

